pkg/ds/farm.go

```diff
@@ -614,29 +625,37 @@ func (dsf *Farm) spawnDaemonSet(
 // 3) If applicable, ensures the worker for this daemon set has the latest copy of the daemon
 // set
 func (dsf *Farm) lockAndSpawn(ctx context.Context, dsFields ds_fields.DaemonSet) bool {
-	var dsUnlocker consul.Unlocker
 	var err error
 	dsLogger := dsf.makeDSLogger(dsFields)
 
+	var unlocker consul.TxnUnlocker
+
 	// If it is not in our map, then try to acquire the lock
 	child, ok := dsf.children[dsFields.ID]
 	if !ok {
-		dsUnlocker, err := dsf.dsLocker.LockForOwnership(dsFields.ID, dsf.session)
-		if _, ok := err.(consul.AlreadyLockedError); ok {
-			// Lock was either already acquired by another farm or it was Acquired
-			// by this farm
-			dsf.logger.Infof("Lock on daemon set '%v' was already acquired by another ds farm", dsFields.ID)
-			if err != nil {
-				dsf.logger.Infof("Additional ds farm lock errors: %v", err)
-			}
-			dsf.releaseLock(dsUnlocker)
+		lockCtx, lockCancel := transaction.New(ctx)
+		defer lockCancel()
+		unlocker, err = dsf.dsLocker.LockForOwnershipTxn(lockCtx, dsFields.ID, dsf.session)
+		if err != nil {
+			dsf.logger.WithError(err).Errorln("could not build daemon set locking transaction")
 			return false
-		} else if err != nil {
+		}
+
+		ok, resp, err := transaction.Commit(lockCtx, dsf.txner)
+		if err != nil {
 			dsf.logger.WithError(err).Errorf("Could not acquire lock on daemon set '%v'", dsFields.ID)
 			// The session probably either expired or there was probably a network
 			// error, so the rest will probably fail
 			dsf.handleSessionExpiry(dsFields, dsLogger, err)
-			dsf.releaseLock(dsUnlocker)
+			return false
+		}
+		if !ok {
+			// Lock was either already acquired by another farm or it was Acquired
+			// by this farm
+			dsf.logger.Infof("Lock on daemon set '%v' was already acquired by another ds farm: %s", dsFields.ID, transaction.TxnErrorsToString(resp.Errors))
```

Comment:
`Errorf()`

Reply:
I don't think it's really an error, it just means that the daemon set is already claimed by another farm. we expect that to happen often